### DISCUSSION TITLE
Idea: useRefetch hook

### DISCUSSION THREAD

**Discussion by sergiodxa** [remix-run/remix#2920] [Proposals]
If you are building something like a dashboard that needs to be updated from time to time without expecting the user to navigate you have right now the option to use Remix loaders to get the initial data and then React Query (or any similar library) providing that initial data and let RQ use their revalidation strategies to keep the data up to date.

If Remix exposed a way to trigger a refetch this could be solved with smaller implementations of only the revalidation strategy you want to use

Example API

```ts
let refetch = useRefetch() // get a function to refetch all loaders

useInterval(refetch, 5000) // refetch loaders every 5s

useCallOnReconnection(refetch) // refetch if user lost internet and recovered it

useOnTabBecomesActive(refetch) // refetch if the user changed to another tab or app and then back to the app

// also let the user manually refetch
return <button onClick={() => refetch()}>Refetch</button>
```

**Comment by tchak** [comment]
I am pretty sure you can already achieve something like this with a combination of `useSubmit` and a `redirect` to self in the action

**Comment by tchak** [comment]
I have this is my app https://gist.github.com/tchak/bdcba6668610520fe8196747d5b27b4b
